src/extension.ts

```diff
@@ -50,9 +52,21 @@ export function activate(context: vscode.ExtensionContext) {
 	DATASOURCE_TYPES.set('Google Sheets', new GoogleSheetsDataSource(TEMPLATE_NAME));
 	DATASOURCE_TYPES.set('Rest Based', new RestBasedDataSource(TEMPLATE_NAME));
 
+	if (!vscode.workspace.workspaceFolders || vscode.workspace.workspaceFolders.length === 0) {
+		workspaceReady = false;
+	}
+
 	dataVirtProvider = new DataVirtNodeProvider(vscode.workspace.rootPath, context);
 	creatDataVirtView();
 
+	vscode.workspace.onDidChangeWorkspaceFolders( event => {
```

Comment:
as event is unused, we can replace with () as suggested by quickix in VS Code

Reply:
Fixed